server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java

```diff
@@ -945,6 +959,10 @@ private void indexValue(DocumentParserContext context, String value) {
 
         if (value.length() > fieldType().ignoreAbove()) {
             context.addIgnoredField(name());
+            if (context.isSyntheticSource()) {
+                // Save a copy of the field so synthetic source can load it
+                context.doc().add(new Field(originalName(), value, ORIGINAL_FIELD_TYPE));
```

Comment:
If `stored=true` is set on here then we store it twice, right? Is it worth detecting that case and deferring to the 'normal' stored field rather than always using the hidden field?

Reply:
If I did that then fields longer than the limit would start being loadable via stored fields. That feels like something folks could accidentally rely on.